hnn_core/gui/gui.py

```diff
@@ -1330,23 +1340,23 @@ def on_upload_params_change(change, params, tstop, dt, log_out, drive_boxes,
 
     ext = Path(params_fname).suffix
     read_func = {'.json': _read_json, '.param': _read_legacy_params}
-    params_network = read_func[ext](param_data)
+    params_network = Params(read_func[ext](param_data))
 
     # update simulation settings and params
-    log_out.clear_output()
     with log_out:
         if 'tstop' in params_network.keys():
             tstop.value = params_network['tstop']
         if 'dt' in params_network.keys():
             dt.value = params_network['dt']
 
-        params.update(params_network)
     # init network, add drives & connectivity
     if load_type == 'connectivity':
+        params.update(params_network)
         add_connectivity_tab(params, connectivity_out, connectivity_textfields)
     elif load_type == 'drives':
-        add_drive_tab(params, drives_out, drive_widgets, drive_boxes, tstop,
-                      layout)
+        with log_out:
+            add_drive_tab(params_network, log_out, drives_out, drive_widgets, drive_boxes, tstop,
```

Comment:
@dylansdaniels for the output to print inside `add_drives_tab` we need to provide `log_out` *both* to the function and wrap it inside the context manager. I think the reason is that `log_out` stops existing once we are out of the `with` block. We need to handle the logging behavior more consistently and cleanly as mentioned [here](https://github.com/jonescompneurolab/hnn-core/issues/737#issuecomment-2164258292)